code/game/objects/structures/watercloset.dm

```diff
@@ -578,6 +578,16 @@
 		the_plate.clean = TRUE
 		O.update_icon()
 
+	if(istype(O, /obj/item/toy/waterballoon))
+		var/obj/item/toy/waterballoon/waterB = O
+		if(waterB.reagents.total_volume >= waterB.reagents.maximum_volume)
+			to_chat(user, "<span class='warning'>\The [waterB] is full.</span>")
+			return
+		waterB.reagents.add_reagent(WATER, min(10 - waterB.reagents.total_volume, 10))
```

Comment:
This is an unnecessary operation since add_reagent already covers this

Reply:
ah, alright.

Reply:
This is kind of the opposite of what I would suggest doing. I think it's valid to keep the "unnecessary" full check so you can get feedback. I'm saying you don't need to do the min() operation because add_reagent won't let you overfill